app/scripts/controllers/list_proposal.js

```diff
@@ -8,19 +8,58 @@
  * Controller of the admissionSystemApp
  */
 angular.module('admissionSystemApp')
-  .controller('ListProposalCtrl', ['$scope','ngTableParams', 'ListProposalGettingService', function ($scope, ngTableParams, ListProposalGettingService) {
+  .controller('ListProposalCtrl', ['$scope','$filter','ngTableParams', 'ListProposalGettingService', '$modal', function ($scope, $filter, ngTableParams, ListProposalGettingService, $modal) {
+
+    $scope.headers = [
+      {name: "id", display: "id", visible: true},
+      {name: "specialtyId", display: "speciality", visible: true},
+      {name: "departmentId", display: "department", visible: true},
+      {name: "timePeriodId", display: "timePeriod", visible: true},
+      {name: "timePeriodCourseId", display: "timePeriodCarousel", visible: true},
+      {name: "specofferTypeId", display: "specofferType", visible: true},
+      {name: "eduFormTypeId", display: "eduFormType", visible: true},
+      {name: "licCount", display: "licCount", visible: true},
+      {name: "stateCount", display: "stateCount", visible: true}
+    ];
+
+    $scope.open = function (size) {
+
+      var modalInstance = $modal.open({
+        templateUrl: '../views/modal/modalFilter.html',
+        controller: function ($scope, $modalInstance) {
+          $scope.headersLocal = $scope.headers;
+          $scope.cancel = function () {
+            $scope.headers = $scope.headersLocal;
+            $modalInstance.dismiss('cancel');
```

Comment:
can you try without this weird hack? just work with $scope.headers.

If it will not work, and you have some free time - you can try to understand why. :)